## Functions

Muestran algo en pantalla 

Devuelven un valor: return

In [1]:
def saludo(): # Define una función
    print("Hola Mundo")

In [2]:
saludo()

Hola Mundo


In [3]:
def tu_saludo(nombre):
    print("Hola " + nombre)

In [6]:
tu_saludo("Juana")

Hola Juana


In [7]:
def suma(a,b):
    res = a+b
    return res

In [10]:
suma(1,2)
# guardando el valor anterior en variable:
resultado = suma(1, 2)
print(resultado)

3


In [11]:
# Función que le pide al usuario un número y devuelve todos los números pares desde ese número hasta 1000
def get_even_numbers(start_number):
  even_numbers = []
  for i in range(start_number, 1001):
    if i % 2 == 0:
      even_numbers.append(i)
  return even_numbers

In [24]:
start_number = int(input())
get_even_numbers(start_number)

[90,
 92,
 94,
 96,
 98,
 100,
 102,
 104,
 106,
 108,
 110,
 112,
 114,
 116,
 118,
 120,
 122,
 124,
 126,
 128,
 130,
 132,
 134,
 136,
 138,
 140,
 142,
 144,
 146,
 148,
 150,
 152,
 154,
 156,
 158,
 160,
 162,
 164,
 166,
 168,
 170,
 172,
 174,
 176,
 178,
 180,
 182,
 184,
 186,
 188,
 190,
 192,
 194,
 196,
 198,
 200,
 202,
 204,
 206,
 208,
 210,
 212,
 214,
 216,
 218,
 220,
 222,
 224,
 226,
 228,
 230,
 232,
 234,
 236,
 238,
 240,
 242,
 244,
 246,
 248,
 250,
 252,
 254,
 256,
 258,
 260,
 262,
 264,
 266,
 268,
 270,
 272,
 274,
 276,
 278,
 280,
 282,
 284,
 286,
 288,
 290,
 292,
 294,
 296,
 298,
 300,
 302,
 304,
 306,
 308,
 310,
 312,
 314,
 316,
 318,
 320,
 322,
 324,
 326,
 328,
 330,
 332,
 334,
 336,
 338,
 340,
 342,
 344,
 346,
 348,
 350,
 352,
 354,
 356,
 358,
 360,
 362,
 364,
 366,
 368,
 370,
 372,
 374,
 376,
 378,
 380,
 382,
 384,
 386,
 388,
 390,
 392,
 394,
 396,
 398,
 400,
 402,
 404,
 406,
 408,
 410,
 412,
 414,
 416,
 418,
 420,
 422,
 42

In [37]:
# Todos los números pares desde 1 hasta el número ingresado por el usuario
def pares(numero): # Define una función llamada 'pares' que toma un parámetro 'numero'
    for i in range(2, numero + 1, 2): # Itera desde 0 hasta el número ingresado por el usuario, incrementando de 2 en 2
        print(i) # Imprime los números pares desde 0 hasta el número ingresado por el usuario


In [36]:
numero = int(input("Ingrese un número: "))  # Solicita al usuario que ingrese un número
pares(numero)  # Llama a la función pares con el número ingresado por el usuario

2
4
6
8
10
12
14
16
18
20


* Ejercicio Funciones en Python 1

Crea una función llamada imprimir_mensaje que, cuando se llame, imprima el mensaje:

"¡Estoy aprendiendo funciones en Python!"

Llama a la función para verificar que imprime el mensaje correctamente.

In [38]:
def imprimir_mensaje(mensaje):
    print(mensaje)

In [39]:
imprimir_mensaje("¡Estoy aprendiendo funciones en Python!")

¡Estoy aprendiendo funciones en Python!


* Ejercicio Funciones en Python 2

Define una función llamada saludar_usuario que acepte un parámetro nombre y luego imprima el mensaje "Hola, {nombre}!", donde {nombre} debe ser reemplazado por el valor del parámetro. Llama a la función pasando tu propio nombre como argumento para probarla.

In [41]:
def saludar_usuario(nombre):
    print(f"Hola, {nombre}!")

In [42]:
saludar_usuario("Yeymi")

Hola, Yeymi!


* Ejercicio Funciones en Python 3

Crea una función llamada calcular_area_rectangulo que acepte dos parámetros: largo y ancho. La función debe calcular el área del rectángulo (largo * ancho) y devolver el resultado. Llama a la función con valores de tu elección y imprime el resultado.

In [43]:
def calcular_area_rectangulo(largo, ancho):
    area = largo * ancho
    return area

In [44]:
print (calcular_area_rectangulo(3,5))

15


## Funciones que llaman funciones

In [45]:
def pedir_nombre():
    nombre = input("Ingrese su nombre: ")
    return nombre

In [46]:
def pedir_apellido():
    apellido = input("Ingrese su apellido: ")
    return apellido

In [47]:
def saludar():
    saludo = f"Hola {pedir_nombre()} {pedir_apellido()}"
    return saludo

In [48]:
saludar()

'Hola Yeymi González'

In [58]:
# Función llamada desde un loop
mi_lista = [1,2,3,4,5]

def cuadrado(n):
    return n**2

def calcular_lista(numeros):
    for numero in numeros:
        resultado = cuadrado(numero)
        print(f"El resultado de {numero} es {resultado}")

In [59]:
calcular_lista(mi_lista)

El resultado de 1 es 1
El resultado de 2 es 4
El resultado de 3 es 9
El resultado de 4 es 16
El resultado de 5 es 25


* Ejercicio Funciones que Llaman Funciones en Python 1

Crea una función llamada solicitar_nombre() que pida al usuario su nombre utilizando input() y retorne este nombre.

Luego, crea otra función llamada saludar() que llame a solicitar_nombre()  e imprima el siguiente saludo (reemplazando [nombre] por el nombre elegido por el usuario):

In [61]:
def solicitar_nombre():
    nombre = input("Ingrese su nombre: ")
    return nombre

In [65]:
def saludar():
    print(f"¡Hola {solicitar_nombre()}!")

In [66]:
saludar()

¡Hola Yeymiiiiiiiiiii!


* Ejercicio Funciones que Llaman Funciones en Python 2

Escribe dos funciones:

pedir_numeros() debe solicitar al usuario que ingrese dos números (uno a la vez) y retornar estos números como enteros.

La segunda función, sumar(), debe llamar a pedir_numeros(), sumar los dos números obtenidos e imprimir el resultado de la suma.

Asegúrate de que sumar() imprima el resultado en formato de texto, por ejemplo:

"La suma es 10" para una suma de 5 y 5.

Para ello puedes usar una cadena literal del tipo f"La suma es {resultado}"  para imprimir el mensaje.

In [72]:
def pedir_numeros():
    num1 = int(input(f"ingrese un númeroo número: "))
    num2 = int(input(f"ingrese otro número: "))
    return num1, num2

In [91]:
def sumar():
    suma = pedir_numeros()
    print(f"La suma de los números ingresados es: {suma[0]+suma[1]}")
    # print(f"El resultado de la suma es: {resultado}")

In [92]:
sumar()

La suma de los números ingresados es: 30


* Ejercicio Funciones que Llaman Funciones en Python 3

Desarrolla un programa que incluya las siguientes funciones:

Función 1:

solicitar_numero() que pide al usuario ingresar un número entero positivo y lo retorna. Puedes usar un mensaje de input del tipo: input("Ingrese un número entero positivo: ")

Recuerda convertir la entrada de usuario a un tipo de dato entero. utiliza int()


Función 2:

calcular_factorial(n) que recibe el número entero n   (devuelto por solicitar_numero() y retorna su factorial.

El factorial de un número entero positivo n se define como el producto de todos los números enteros positivos desde 1 hasta n.

Por ejemplo, el factorial de 5 (simbolizado como 5) es 1 * 2 * 3 * 4 * 5 = 120. 

In [115]:
def solicitar_numero():
    return int(input("Ingrese un número entero positivo: "))

def calcular_factorial(n):
    if n == 0:
        print("El factorial de 0 es 1")
    else:
        factorial = 1
        for i in range(1, n + 1):
            factorial *= i
        return factorial

In [116]:
def mostrar_resultado():
    n = solicitar_numero()
    factorial = calcular_factorial(n)
    print(f"El factorial de {n} es {factorial}")

In [118]:
mostrar_resultado()

El factorial de 5 es 120


In [ ]:
def solicitar_numero():
    return int(input("Ingrese un número entero positivo: "))
def calcular_factorial(n):
    factorial = 1
    for i in range(1, n+1):
        factorial *= i
    return factorial
def mostrar_resultado():
    n = solicitar_numero()
    resultado = calcular_factorial(n) 
    print(f"El factorial de {n} es {resultado}")